python/ray/data/dataset.py

```diff
@@ -993,29 +996,37 @@ def union(self, *other: List["Dataset[T]"]) -> "Dataset[T]":
         """
 
         start_time = time.perf_counter()
-        context = DatasetContext.get_current()
-        tasks: List[ReadTask] = []
-        block_partition_refs: List[ObjectRef[BlockPartition]] = []
-        block_partition_meta_refs: List[ObjectRef[BlockPartitionMetadata]] = []
 
         datasets = [self] + list(other)
+        bls = []
+        has_nonlazy = False
         for ds in datasets:
             bl = ds._plan.execute()
-            if isinstance(bl, LazyBlockList):
+            if not isinstance(bl, LazyBlockList):
+                has_nonlazy = True
+            bls.append(bl)
```

Comment:
Based on implementation, I think the API name is wrong - this is really concat not union.

Reply:
Hmm that's true, good API feedback. @ericl what do you think about me changing `.union()` to `.concat()` in another PR?

Reply:
How do we rename a public API? It looks tricky as people somewhere may already use it.